### PR TITLE
fix(typescript): migrate InputGroup test and stories to typescript and improve InputGroup types

### PR DESCRIPTION
**Pull request by ivanhdez28**
- Migrate leftover InputGroup javascript files to typescript.
- Improve InputGroup props types to better fit with `rest` parameters.

### CODE REVIEW

File: src/components/InputGroup/index.test.tsx [hunk 1]
```diff
@@ -21,19 +21,19 @@ describe('InputGroup component', () => {
...
     });
 
     it('sets expected default for 3 items', () => {
-      expect(getFlexSettings(3)).toStrictEqual(['shrink', 'grow', 'shrink']);
+      expect(getFlexSettings(3, [])).toStrictEqual(['shrink', 'grow', 'shrink']);
     });
 
     it('sets expected default for 4 items', () => {
-      expect(getFlexSettings(4)).toStrictEqual(['shrink', 'grow', 'grow', 'shrink']);
+      expect(getFlexSettings(4, [])).toStrictEqual(['shrink', 'grow', 'grow', 'shrink']);
```
Comment (reywright) [inline src/components/InputGroup/index.test.tsx:32]: Is there a change that would make it not necessary to repeatedly pass an empty array to this function? Whenever I see anything like this, it's a smell for something else that probably needs to be changed.

File: src/components/InputGroup/index.tsx [hunk 2]
```diff
@@ -12,7 +12,10 @@ type FlexType = 'shrink' | 'grow'[];
  * @param {Array} userSettings - `flexSettings` prop optionally set by user
  * @returns {Array} e.g. ['shrink', 'grow', 'shrink']
  */
-export const getFlexSettings = (childCount: number, userSettings: FlexType): FlexType => {
+export const getFlexSettings = (
+  childCount: number,
+  userSettings: FlexType[]
```
Comment (reywright) [inline src/components/InputGroup/index.tsx:17]: if `flexSettings` below is optional, and we're passing it into `getFlexSettings` as `userSettings`... then I think `userSettings` should also be optional, which would remove the need to pass empty arrays into this function.

**Review by reywright — CHANGES_REQUESTED** [review]
Thanks 🙏 Some questions on this PR.

File: src/components/InputGroup/index.test.tsx (hunk 1 above)
Comment (ivanhdez28) [reply src/components/InputGroup/index.test.tsx:32]: we can change the 2nd param to be optional

File: src/components/InputGroup/index.tsx (hunk 2 above)
Comment (ivanhdez28) [reply src/components/InputGroup/index.tsx:17]: agree

File: src/components/InputGroup/index.test.tsx [hunk 3]
```diff
@@ -21,19 +21,19 @@ describe('InputGroup component', () => {
 
   describe('getFlexSettings', () => {
     it('sets expected default for 2 items', () => {
-      expect(getFlexSettings(2)).toStrictEqual(['grow', 'shrink']);
+      expect(getFlexSettings(2, [])).toStrictEqual(['grow', 'shrink']);
     });
 
     it('sets expected default for 3 items', () => {
-      expect(getFlexSettings(3)).toStrictEqual(['shrink', 'grow', 'shrink']);
+      expect(getFlexSettings(3, [])).toStrictEqual(['shrink', 'grow', 'shrink']);
```
[the remark this answers is not in the document]
Comment (ivanhdez28) [reply src/components/InputGroup/index.test.tsx:28]: my take on this is that its type says it's an array of `'grow' | 'shrink'`, just that it holds none at that moment, just like a `number[]` can be empty too.

`const nums: number[] = []` is fine as well.

[example](https://www.typescriptlang.org/play?#code/MYewdgzgLgBGCuBbCAuOSBGBTATgbQF0YBeGQgbiA).

File: src/components/TextInput/index.stories.tsx [hunk 4]
```diff
@@ -10,7 +10,7 @@ import TextInput, { TextInputProps } from '.';
 import { noop } from 'util/index';
 
 export const Primary = (args: TextInputProps): JSX.Element => (
-  <TextInput {...args} label="First Name" />
+  <TextInput {...args} label="First Name" ref={null} />
```
[the remark this answers is not in the document]
Comment (ivanhdez28) [reply src/components/TextInput/index.stories.tsx:13]: what comes from the generic `DetailedHTMLProps` is a `LegacyRef<T>` which can be `string | RefObject<T>`, i specified `ref?: RefObject<HTMLInputElement>` in `TextInputProps` so we don't have to do `LegacyRef` nor assign null here, sorry i didn't look for a better solution from the beginning.

**Review by David-Cohen35 — APPROVED** [review]
looks good

File: src/components/TextInput/index.stories.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (ivanhdez28) [reply src/components/TextInput/index.stories.tsx]: i did some research and if we extend from any generic that react provides, refs will be legacy, understandable for backwards compatibility and support, so the solution i'd suggest here is to redefine in the interface as `ref?: RefObject<HTMLInputElement>;`

from react types:
```
type Ref<T> = RefCallback<T> | RefObject<T> | null;
type LegacyRef<T> = string | Ref<T>;
```

`RefObject` falls under `LegacyRef` so we're not breaking anything.

File: src/components/InputGroup/index.test.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (reywright) [reply src/components/InputGroup/index.test.tsx]: ah. Makes sense.

**Review by reywright — APPROVED** [review]
Looks good! Thanks for making adjustments here.

**Pull request merged**